assignments/prototype-refactor.js

```diff
@@ -4,6 +4,133 @@ Prototype Refactor
 
 1. Copy and paste your code or the solution from yesterday
 
+function Gameobject(attr) {
+  this.createdAt = attr.createdAt;
+  this.dimensions = {
+    length: 0,
+    width: 0,
+    height: 0
+  };
+}
+
+Gameobject.prototype.destroy = function() {
+  return "Object was removed from game";
+};
+
+Characterstats.prototype = Object.create(Gameobject.prototype);
+function Characterstats(attr) {
+  Gameobject.call(this, attr);
+  this.name = attr.name;
+  this.healthPoints = attr.healthPoints;
+}
+
+Characterstats.prototype.takeDamage = function() {
+  return `${this.name} took damage.`;
+};
+
+Humanoid.prototype = Object.create(Characterstats.prototype);
+
+function Humanoid(attrs) {
+  Characterstats.call(this, attrs);
+  this.team = attrs.team;
+  this.weapons = attrs.weapons;
+  this.language = attrs.language;
+}
+
+Humanoid.prototype.greet = function() {
+  return `${this.name} offers a greeting in ${this.language}.`;
+};
 2. Your goal is to refactor all of this code to use ES6 Classes. The console.log() statements should still return what is expected of them.
 
 */
+
+class Gameobject {
+  constructor(attributes) {
+    this.createdAt = attributes.createdAt;
+    this.dimensions = {
+      length: 0,
+      height: 0,
+      weight: 0
+    };
+  }
+  destroy() {
+    return "Object was removed from game";
+  }
+}
+
+class Characterstats extends Gameobject {
+  constructor(attributes) {
+    super(attributes);
+    this.name = attributes.name;
+    this.healthpoints = attributes.healthpoints;
```

Comment:
watch your camel casing here. variable names are case sensitive so you have to be consistent.